apps/calendar/js/views/modify_event.js

```diff
@@ -357,114 +256,69 @@ ModifyEvent.prototype = {
       return;
     }
 
-    var self = this;
-    var provider;
+    // mark view as 'in progress' so we can style
+    // it via css during that time period
+    this.element.classList.add(this.PROGRESS);
 
-    this.store.providerFor(this.event, fetchProvider);
+    var moveDate = this.event.startDate;
 
-    function fetchProvider(err, result) {
-      provider = result;
-      provider.eventCapabilities(
-        self.event.data,
-        verifyCaps
-      );
-    }
+    try {
+      yield core.bridge[method](this.event);
```

Comment:
Do you think it might make sense to move some of this into a helper function? This is a lot of logic for a single try/catch block